lib/WebService/Async/Segment/Customer.pm

```diff
@@ -0,0 +1,198 @@
+package WebService::Async::Segment::Customer;
+
+use strict;
+use warnings;
+
+use constant COMMON_FIELDS => qw(userId anonymousId traits context integrations timestamp);
+
+# VERSION
+
+=head1 NAME
+
+WebService::Async::Segment::Customer - represents a customer object with methods to make Segment API calls.
+
+=head1 DESCRIPTION
+
+You can create object of this class directly or indirectly using B<WebService::Async::Segment::new_customer>.
+It is possible to B<identify> and B<track> calls with corresponding objects method created in this module.
+
+=cut
+
+=head1 METHODS
+=head2 new
+
+Class constructor accepting a hash of named args containing customer info, along with a Segment API wrapper object (an object of class C<WebService::Async::Segment::Customer>.
+The accepted params are:
+
+=over 4
+
+=item * C<api_client> - Segment API wrapper object.
+
+=item * C<userId> - Unique identifier of a user.
+
+=item * C<anonymousId> - A pseudo-unique substitute for a User ID, for cases when you don't have an absolutely unique identifier.
+
+=item * C<traits> - Free-form dictionary of traits of the user, like email or name.
+
+=back
+
+=cut
+
+sub new {
+    my ($class, %args) = @_;
+
+    Scalar::Util::weaken($args{api_client}) if $args{api_client};
+
```

Comment:
Do we  keep `WebService::Async::Segment::Customer` objects  inside `api_client` object? (I didn't find it). why do we need to use `weaken` here?

Reply:
Sounds reasonable. I removed the line.